This patch fixes an issue where pressing any key within the gummiboot
menu causes it to hang (though no issue occurs if no key is pressed).  The
culprit appears to be the WaitForKeyEx function, which lacks support in certain
UEFI implementations.  This fix is backported from the same issue found in
systemd-boot (gummiboot's mainstream counterpart): see
<https://github.com/systemd/systemd/issues/3632> and
<https://github.com/systemd/systemd/pull/3735>.

--- a/src/efi/console.c	2015-03-11 16:50:35.000000000 -0700
+++ b/src/efi/console.c	2020-12-09 15:41:19.837704801 -0800
@@ -96,10 +96,7 @@
 
         /* wait until key is pressed */
         if (wait) {
-                if (TextInputEx)
-                        uefi_call_wrapper(BS->WaitForEvent, 3, 1, &TextInputEx->WaitForKeyEx, &index);
-                else
-                        uefi_call_wrapper(BS->WaitForEvent, 3, 1, &ST->ConIn->WaitForKey, &index);
+                uefi_call_wrapper(BS->WaitForEvent, 3, 1, &ST->ConIn->WaitForKey, &index);
         }
 
         if (TextInputEx) {
